cli/src/data.ts

```diff
@@ -59,25 +66,79 @@ function loadTinderDogData(split: number): Dataset<DataFormat.Raw["image"]> {
     });
 }
 
+async function readLabels(labelsPath:string): Promise<string[]>{
+  const raw = await fs.readFile(labelsPath, "utf8");
+  return raw.split(/\r?\n/).map(s => s.trim()).filter(s => s.length > 0);
+}
+
+function loadExtCifar10(userIdx: number){
+  const CIFAR10_LABELS = ["airplane", "automobile", "bird", "cat", "deer", "dog", "frog", "horse", "ship", "truck"];
+  const folder = path.join("..", "datasets", "extended_cifar10");
+  const clientFolder = path.join(folder, `client_${userIdx}`);
+
+  return new Dataset<[Image, string]>(async function*(){
+    const entries = await fs.readdir(clientFolder, {withFileTypes: true});
+
+    const items = entries
+        .map((e) => {
+          const m = e.name.match(
+            /^image_(\d+)_label_(\d+)\.png$/i
+          );
+          if (!m) return null;
```

Comment:
this `map(…).filter(…)` is quite common and has, in most languages, a nicer way to achieved it (and reduce the runtime cost of double trasversal). in js, it is `flatMap` where you can return an array of valid values

```suggestion
        .flatMap((e) => {
          const m = e.name.match(
            /^image_(\d+)_label_(\d+)\.png$/i
          );
          if (m === null) return [];
```